corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -424,7 +425,77 @@ hqDefine("geospatial/js/geospatial_map", [
 
         var missingGPSModel = function(cases) {
             this.casesWithoutGPS = ko.observable(cases);
+        var userFiltersModel = function () {
+            var self = {};
+
+            self.shouldShowUsers = ko.observable(false);
+            self.hasFiltersChanged = ko.observable(false);  // Used to disable "Apply" button
+            self.showFilterMenu = ko.observable(true);
+            self.hasErrors = ko.observable(false);
+
+            self.loadUsers = function () {
+                map.removeUserMarkers();
+                self.hasErrors(false);
+                if (!self.shouldShowUsers()) {
+                    self.hasFiltersChanged(false);
+                    return;
+                }
+
+                $.ajax({
+                    method: 'GET',
+                    url: initialPageData.reverse('get_users_with_gps'),
+                    success: function (data) {
+                        self.hasFiltersChanged(false);
+
+                        const usersWithoutGPS = data.user_data.filter(function (item) {
+                            return item.gps_point === null || !item.gps_point.length;
+                        });
+                        missingGPSModelInstance.usersWithoutGPS(usersWithoutGPS);
+
+                        const usersWithGPS = data.user_data.filter(function (item) {
+                            return item.gps_point !== null && item.gps_point.length;
+                        });
+
+                        const userData = _.object(_.map(usersWithGPS, function (dataItem) {
+                            const gpsData = (dataItem.gps_point) ? dataItem.gps_point.split(' ') : [];
+                            const lat = parseFloat(gpsData[0]);
+                            const lng = parseFloat(gpsData[1]);
+
+                            const editUrl = initialPageData.reverse('edit_commcare_user', dataItem.id);
+                            const link = `<a class="ajax_dialog" href="${editUrl}" target="_blank">${dataItem.username}</a>`;
+
+                            return [dataItem.id, {'coordinates': {'lat': lat, 'lng': lng}, 'link': link}];
+                        }));
+
+                        map.addUserMarkersToMap(userData, defaultUserMarkerColor);
+                    },
+                    error: function () {
+                        self.hasErrors(true);
+                    },
+                });
+            };
+
+            self.onFiltersChange = function () {
+                self.hasFiltersChanged(true);
+            };
+
+            self.toggleFilterMenu = function () {
+                self.showFilterMenu(!self.showFilterMenu());
+                const shouldShow = self.showFilterMenu() ? 'show' : 'hide';
+                $("#user-filters-panel .panel-body").collapse(shouldShow);
```

Comment:
What is happening here?
`collapse` needs a string to be passed?

Reply:
This toggles collapsing/showing the user filter panel, similar to how the case filters menu works. According to the [Bootstrap documentation](https://getbootstrap.com/docs/3.4/javascript/#collapse-options), a string must be passed to `collapse()`.

If we want to simplify this a little, we could change it to `$("#user-filters-panel .panel-body").collapse('toggle')` but I think keeping it explicit as it is now is also fine.